sysboundary/outflow.cpp

```diff
@@ -420,16 +356,27 @@ namespace SBC {
       
       const OutflowSpeciesParameters& sP = this->speciesParams[popID];
       SpatialCell* cell = mpiGrid[cellID];
+      if (cell->sysBoundaryFlag != this->getIndex()) {
+         return;
+      }
+
       creal* const cellParams = cell->parameters.data();
-      creal dx = cellParams[CellParams::DX];
-      creal dy = cellParams[CellParams::DY];
-      creal dz = cellParams[CellParams::DZ];
+      Real dx = cellParams[CellParams::DX];
+      Real dy = cellParams[CellParams::DY];
+      Real dz = cellParams[CellParams::DZ];
       creal x = cellParams[CellParams::XCRD] + 0.5*dx;
       creal y = cellParams[CellParams::YCRD] + 0.5*dy;
       creal z = cellParams[CellParams::ZCRD] + 0.5*dz;
       
       bool isThisCellOnAFace[6];
-      determineFace(&isThisCellOnAFace[0], x, y, z, dx, dy, dz, true);
+      // if refLevel isn't 0, assume neighbour might be on a lower refinement level
+      if (cellParams[CellParams::REFINEMENT_LEVEL] > 0) {
+         dx *= 2;
+         dy *= 2;
+         dz *= 2;
+      }
+
+      determineFace(&isThisCellOnAFace[0], x, y, z, dx*2, dy*2, dz*2, true);
```

Comment:
I don't understand this double *2, and shouldn't it be /2 anyway or?... Anyway, you got me confused, please explain.

Reply:
I found it more reasonable to define dx in `determineFace` as the range from edge to count as being on a face. I suppose this just moves `*2` around though.